src/main/kotlin/com/example/iotmanagementdevice/repository/UserRepository.kt

```diff
@@ -1,9 +1,23 @@
 package com.example.iotmanagementdevice.repository
 
-import com.example.iotmanagementdevice.model.User
-import org.springframework.data.jpa.repository.JpaRepository
+import com.example.iotmanagementdevice.model.MongoDevice
+import com.example.iotmanagementdevice.model.MongoUser
+import org.bson.types.ObjectId
 
-interface UserRepository : JpaRepository<User, Long> {
-    fun findByName(name: String): User?
-    fun findByEmail(email: String): User?
+interface UserRepository {
+    fun findById(id: ObjectId): MongoUser?
+
+    fun findAll(): List<MongoUser>
+
+    fun assignDeviceToUser(userId: ObjectId, deviceId: ObjectId): Boolean
+
+    fun save(user: MongoUser): MongoUser?
+
+    fun deleteById(id: ObjectId)
+
+    fun findByUserName(username: String): MongoUser?
+
+    fun findByUserEmail(email: String): MongoUser?
+
+    fun findDevicesByUserId(userId: ObjectId): List<MongoDevice>?
```

Comment:
Personally, i would transfer this to a DeviceRepository.
Kinda strange expecting List<MongoDevice> in _User_ repository
Btw, return type in your implementation is not nullable